app/src/sharedTest/java/org/oppia/android/app/home/HomeFragmentViewModelsTest.kt

```diff
@@ -0,0 +1,1233 @@
+import org.oppia.android.domain.topic.PrimeTopicAssetsControllerModule
+import org.oppia.android.testing.RobolectricModule
+import org.oppia.android.testing.TestAccessibilityModule
+import org.oppia.android.testing.TestDispatcherModule
+import org.oppia.android.testing.TestLogReportingModule
+import org.oppia.android.util.caching.testing.CachingTestModule
+import org.oppia.android.util.gcsresource.GcsResourceModule
+import org.oppia.android.util.logging.LoggerModule
+import org.oppia.android.util.logging.firebase.FirebaseLogUploaderModule
+import org.oppia.android.util.parser.GlideImageLoaderModule
+import org.oppia.android.util.parser.HtmlParserEntityTypeModule
+import org.oppia.android.util.parser.ImageParsingModule
+import org.oppia.android.util.system.OppiaClock
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+
+// Time: Wed Apr 24 2019 08:22:00
+private const val MORNING_TIMESTAMP = 1556094120000
+
+// Time: Tue Apr 23 2019 23:22:00
+private const val EVENING_TIMESTAMP = 1556061720000
+
+/** Tests for [HomeViewModel]s data. */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(
+  application = HomeFragmentViewModelsTest.TestApplication::class,
+  manifest = Config.NONE
+)
+class HomeFragmentViewModelsTest {
+  @Inject
+  lateinit var context: Context
+
+  private var morningClock = OppiaClock()
+  private var eveningClock = OppiaClock()
+  private val promotedStory1 = PromotedStory.newBuilder()
+    .setStoryId("id_1")
+    .setStoryName("Story 1")
+    .setTopicName("topic_name")
+    .setTotalChapterCount(1)
+    .build()
+  private val promotedStory2 = PromotedStory.newBuilder()
+    .setStoryId("id_2")
+    .setStoryName("Story 2")
+    .setTopicName("topic_name")
+    .setTotalChapterCount(1)
+    .build()
+  private val promotedStory3 = PromotedStory.newBuilder()
+    .setStoryId("id_3")
+    .setStoryName("Story 3")
+    .setTopicName("topic_name")
+    .setTotalChapterCount(1)
+    .build()
+  private val topicSummary1 = TopicSummary.newBuilder()
+    .setTopicId("id_1")
+    .setName("topic_name")
+    .setTotalChapterCount(2)
+    .build()
+  private val topicSummary2 = TopicSummary.newBuilder()
+    .setTopicId("id_2")
+    .setName("topic_name")
+    .setTotalChapterCount(2)
+    .build()
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+    setUpClocks()
+  }
+
+  private fun setUpTestApplicationComponent() {
+    ApplicationProvider.getApplicationContext<TestApplication>().inject(this)
+  }
+
+  private fun setUpClocks() {
+    morningClock.setCurrentTimeMs(MORNING_TIMESTAMP)
+    eveningClock.setCurrentTimeMs(EVENING_TIMESTAMP)
+  }
+
+  private fun getTestFragment(activity: HomeFragmentTestActivity): HomeFragment {
+    return activity.supportFragmentManager.findFragmentByTag(
+      "home_fragment_test_activity"
+    ) as HomeFragment
+  }
```

Comment:
Could we just create a new fragment instead of using HomeFragment? In general, prefer to test these view models in isolation. E.g. I think you might just be able to do:

```kotlin
private val testFragment by lazy { Fragment() }

...

private fun setUpTestFragment(activity: Activity) {
  activity.supportFragmentManager.beginTransaction().add(TEST_FRAGMENT_TAG, testFragment).commitNow()
}

```

This simplifies the test activity and avoids the HomeFragment dependency.

Reply:
Done -- removed transaction from the test activity as well